Azure/azure-cli
# [Keyvault] BREAKING CHANGE: Fix #10254 & #18479: `az keyvault network-rule add`

**Description**

Fix #10254 
`--ip-address` previously only supports one value. 
Add capability to accept a list of ip addresses. 

Fix #18479 
`--ip-address` previously allow adding an overlapping ip-address without CIDR suffix.
For example, there is an existing ip-address rule of 1.2.3.4/32
`az keyvault network-rule add --ip-address 1.2.3.4` would still add another rule entry.

**Testing Guide**

Fix #10254 
`az keyvault network-rule add --ip-address 1.2.3.4/32 1.2.3.5 -n {keyvault name}`
Fix #18479 
`az keyvault network-rule add --ip-address 1.2.3.4/32 -n {keyvault name}`
`az keyvault network-rule add --ip-address 1.2.3.4 -n {keyvault name}` should give a warning and not add another rule.

**History Notes**


[Keyvault] Fix #10254 : `az keyvault network-rule add --ip-address`: Add capability to accept multiple ip-addresses as a list in the form of `--ip-address ip1 [ip2] [ip3]...`
[Keyvault] BREAKING CHANGE: Fix #18479 : `az keyvault network-rule add --ip-address`: Fix the bug which allows duplicate ip-address with the ones already in the network-rule

---
This checklist is used to make sure that common guidelines for a pull request are followed.
